packages/plugin-compiler/src/main/kotlin/io/realm/compiler/IrUtils.kt

```diff
@@ -107,8 +109,20 @@ fun IrPluginContext.blockBody(
 val ClassDescriptor.isRealmObjectCompanion
     get() = isCompanionObject && (containingDeclaration as ClassDescriptor).hasRealmModelInterface
 
-val ClassDescriptor.hasRealmModelInterface
-    get() = getSuperInterfaces().firstOrNull { it.fqNameSafe == FqNames.REALM_MODEL_INTERFACE } != null
+val ClassDescriptor.hasRealmModelInterface: Boolean
+    get() {
+        // Using PSI to find super types to avoid cyclic reference (see https://github.com/realm/realm-kotlin/issues/339)
+        var hasRealmObjectAsSuperType = false
+        this.findPsi()?.acceptChildren(object : PsiElementVisitor() {
```

Comment:
When could `findPsi()` be false? Should this be considered a fatal error instead?

Reply:
see https://github.com/realm/realm-kotlin/pull/845#discussion_r881049640